src/ripple/app/misc/impl/LoadFeeTrack.cpp

```diff
@@ -150,43 +205,43 @@ scaleFeeLoad(std::uint64_t fee, LoadFeeTrack const& feeTrack,
     // The denominator of the fraction we're trying to compute.
     // fees.units and lftNormalFee are both 32 bit,
     //  so the multiplication can't overflow.
-    auto den = safe_cast<std::uint64_t>(fees.units)
+    auto den = FeeUnit64{ fees.units }
         * safe_cast<std::uint64_t>(feeTrack.getLoadBase());
     // Reduce fee * baseFee * feeFactor / (fees.units * lftNormalFee)
     // to lowest terms.
-    lowestTerms(fee, den);
-    lowestTerms(baseFee, den);
-    lowestTerms(feeFactor, den);
+    lowestTerms2(fee, den);
+    lowestTerms2(baseFee, den);
+    lowestTerms1(feeFactor, den);
 
     // fee and baseFee are 64 bit, feeFactor is 32 bit
     // Order fee and baseFee largest first
-    if (fee < baseFee)
-        std::swap(fee, baseFee);
+    maybe_swap(fee, baseFee);
     // If baseFee * feeFactor overflows, the final result will overflow
-    const auto max = std::numeric_limits<std::uint64_t>::max();
+    constexpr XRPAmount max =
+        std::numeric_limits<XRPAmount::value_type>::max();
     if (baseFee > max / feeFactor)
         Throw<std::overflow_error> ("scaleFeeLoad");
     baseFee *= feeFactor;
     // Reorder fee and baseFee
-    if (fee < baseFee)
-        std::swap(fee, baseFee);
-    // If fee * baseFee / den might overflow...
-    if (fee > max / baseFee)
+    maybe_swap(fee, baseFee);
+    // If fee * baseFee overflows, do the division first
+    if (fee > FeeUnit64{ max / baseFee })
```

Comment:
Do you need this `FeeUnit64` ?

Reply:
This part was rewritten, so technically, no.